src/Tracing.jl

```diff
@@ -201,9 +125,88 @@ function traced_type(::Type{T}, seen, mode, track_numbers) where {T}
     throw(NoFieldMatchError(T, TT2))
 end
 
-function traced_type(
-    ::Type{<:ConcreteRNumber{T}}, seen, ::Val{mode}, track_numbers
-) where {T,mode}
+for T in (DataType, Module, Nothing, Symbol, AbstractChar, AbstractString, AbstractFloat, Integer, RNumber)
+    @eval Base.@nospecializeinfer function traced_type(@nospecialize(T::Type{<:$T}), seen, mode, track_numbers)
```

Comment:
here different `Val(mode)` would provoke different specializations when its actually not used

```suggestion
    @eval Base.@nospecializeinfer function traced_type(@nospecialize(T::Type{<:$T}), seen, @nospecialize(mode), track_numbers)
```